deepchecks/base/check.py

```diff
@@ -332,6 +338,157 @@ def name(cls):
+
+        Args:
+            datasets (List[Dataset]): list of datasets to validate
+
+        Returns:
+            Hashable: name of the label column
+
+        Raises:
+            DatasetValidationError: if datasets do not share same label;
+        """
+        if Dataset.datasets_share_label(datasets) is False:
+            raise DatasetValidationError('Check requires Datasets to have and to share the same label')
+        return cast(Hashable, datasets[0].label_name)
+
+    @staticmethod
+    def _dataset_has_label(dataset: Dataset) -> pd.Series:
+        """Verify that provided dataset has label, otherwise raise an exception.
+
+        Args:
+            datasets (Dataset): dataset to validate
+
+        Returns:
+            pandas.Series: label column
+
+        Raises:
+            DatasetValidationError: if dataset does not have label;
+        """
+        if dataset.label_col is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without label')
+        return dataset.label_col
+
+    @staticmethod
+    def _dataset_has_features(dataset: Dataset) -> pd.DataFrame:
+        """Verify that provided dataset has features, otherwise raise an exception.
+
+        Args:
+            datasets (Dataset): dataset to validate
+
+        Returns:
+            pandas.DataFrame: features dataframe
+
+        Raises:
+            DatasetValidationError: if dataset does not have features;
+        """
+        if (
+            dataset.features_columns is None
+            or len(dataset.features_columns.columns) == 0
+        ):
+            raise DatasetValidationError('Check is irrelevant for Datasets without features')
+        return dataset.features_columns
+
+    @staticmethod
+    def _dataset_has_date(dataset: Dataset) -> pd.Series:
+        """Verify that provided dataset has date column, otherwise raise an exception.
+
+        Args:
+            datasets (Dataset): dataset to validate
+
+        Returns:
+            pandas.Series: date column
+
+        Raises:
+            DatasetValidationError: if dataset does not have date column;
+        """
+        if dataset.datetime_col is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without datetime column')
+        return dataset.datetime_col
+
+    @staticmethod
+    def _dataset_has_index(dataset: Dataset) -> pd.Series:
+        """Verify that provided dataset has index, otherwise raise an exception.
+
+        Args:
+            datasets (Dataset): dataset to validate
+
+        Returns:
+            pandas.Series: dataset index column
+
+        Raises:
+            DatasetValidationError: if dataset does not have index;
+        """
+        if dataset.index_col is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without an index')
+        return dataset.index_col
+
+    @staticmethod
+    def _verify_model_type(
```

Comment:
why it should be in the Check class? I think will be better in utild or dataset class

Reply:
- it is used only within checks to unify error message
- it raises  an exception with check related message